src/entropies/direct_entropies/order_statistics/Vasicek.jl

```diff
@@ -0,0 +1,52 @@
+export Vasicek
+
+"""
+    Vasicek <: IndirectEntropy
+    Vasicek(; m::Int = 1, base = 2)
+
+An indirect entropy estimator used in [`entropy`](@ref)`(Vasicek(), x)` to
+estimate the Shannon entropy of the timeseries `x` to the given
+`base` using the method from Vasicek (1976)[^Vasicek1976].
+
+## Description
+
+The Vasicek entropy estimator first computes the order statistics
+``X_{(1)} \\leq X_{(2)} \\leq \\cdots \\leq X_{(n)}`` for a random sample of length ``n``,
+i.e. the input timeseries. The entropy for the length-`n` sample is then estimated as
+
+```math
+H_V(m, n) =
+\\dfrac{1}{n} \\sum_{i = 1}^n \\log \\left[ \\dfrac{n}{2m} (X_{(i+m)} - X_{(i-m)}) \\right]
+```
```

Comment:
This docstring confused me a bit. What is the returned quantity here? the user provides `m` but not `n`. Where is `n` coming from? Is it the length of the timeseries? Or is it smaller and one computes `H_V(m, n)` for all `n` and then averages?

Reply:
Furthermore, I don't understand what you mean by "order statistics". ``X_{(1)} \\leq X_{(2)} \\leq \\cdots \\leq X_{(n)}`` is a chained boolean expression whose outcome can only be true or false. What are the statistics here? The operation `x < y < z` has no statistics, it is just an expression which evaluates to either true or false.

Reply:
The k-th [order statistics](https://en.wikipedia.org/wiki/Order_statistic) of a random sample is its k-th smallest value. 

It does appear quite a lot in the literature, but I'm fine with skipping technical jargon here and just saying "the k-th smallest sorted value" instead.

Reply:
> This docstring confused me a bit. What is the returned quantity here? the user provides m but not n. Where is n coming from? Is it the length of the timeseries? Or is it smaller and one computes H_V(m, n) for all n and then averages?

I will clarify this.

Reply:
Aaaa okay. I think the solution is to just add a huperlink to the "order statistics" to the wiki page you shared and say in parenthesis (k-th smallest value). Its good to leave order statistics for the search bar.

Reply:
> Aaaa okay. I think the solution is to just add a huperlink to the "order statistics" to the wiki page you shared and say in parenthesis (k-th smallest value). Its good to leave order statistics for the search bar.

Ok, then that's decided.